### DISCUSSION TITLE
Welcome to bhagavad-gita-api Discussions!

### DISCUSSION THREAD

**Discussion by PtPrashantTripathi** [vedicscriptures/bhagavad-gita-api#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tripdog** [comment]
This is awesome. But can somebody please offer me some help to get my request to the API working properly?

**Reply by PtPrashantTripathi** [reply]
**Documentation** : https://documenter.getpostman.com/view/13400094/Tzm5JxDP

**Comment by hemantknayak** [comment]
i have tried to user ur api(free version). But everytime i call the url, 
https://bhagavadgitaapi.in/slok?api_key=***&ch=12&sl=1
 its giving me result from different chapters and different sloks !! Is there any solution? 


![IMG_20210926_163700](https://user-images.githubusercontent.com/26687933/134805159-0be788b8-8a6e-4018-83ec-ea0b34a03f9a.jpg)

**Reply by PtPrashantTripathi** [reply]
@hemantknayak  your are using slok endpoint with wrong parameters
Watch my tutorial then you can understand how to use it.
https://youtu.be/1OVXdkFoKIg?t=3m10s


And here is postman collection link https://documenter.getpostman.com/view/13400094/Tzm5JxDP

**Reply by hemantknayak** [reply]
Thanks for the reply. i was able to get the correct response in postman. But when i am trying to call it from an angular environment, i am getting the cors issue like below.
![Capture](https://user-images.githubusercontent.com/55045579/135007580-11f692af-a548-4d25-853b-993c5a1ebf90.PNG)

**Reply by PtPrashantTripathi** [reply]
I have no idea about angular, but **I don't thinks it's API issue** ,

It's a security policy in angular, 
just google this error you get solution

**Comment by bitwisebro** [comment]
Pranaam @PtPrashantTripathi while consuming the particular slok api, i am facing status code 403, error code: 1010 i.e. "The owner of this website has banned your access based on your browser's signature" . May I get some help please??

**Reply by PtPrashantTripathi** [reply]
> Pranaam @PtPrashantTripathi while consuming the particular slok api, i am facing status code 403, error code: 1010 i.e. "The owner of this website has banned your access based on your browser's signature" . May I get some help please??

Hi @RealGJ 
 Probably it's cause due to some malicious activity? Well we are using cloudflare where I have DDOS protection and "under attack" enabled AFAIK. So that why it's showing you that msg. I'll suggest you try again after some. It will work.

**Comment by bitwisebro** [comment]
Hare krishna @PtPrashantTripathi , recently I've been unable to follow redirects of random slok end point of the API using java 11's httpclient! I've also tried it with unirest and okhttp as shown in postman tutorial link you shared above! Can you help me with that if you have experience in Java??  
I guess same goes for python! I'll let you know once I'm sure! Dhanyawaad!

**Reply by PtPrashantTripathi** [reply]
Looks like this random slok endpoint is causing problem for many people,

I'll suggest you to use this kind of function to get random slok I'd then make the call to api directly to slok

```js
// gitaslokid function
    function gitaslokid() {
        const slokcount = [47, 72, 43, 42, 29, 47, 30, 28, 34, 42, 55, 20, 35, 27, 20, 24, 28, 78]
        const chapter = Math.floor(Math.random() * 17) + 1
        const slok = Math.floor(Math.random() * slokcount[chapter - 1]) + 1
        return `https://bhagavadgitaapi.in/slok/${chapter}/${slok}/`
    }
```

**Reply by bitwisebro** [reply]
Thanks! works for me!

**Comment by mnsdojo** [comment]
Any plan to add endpoint for search ?

**Reply by PtPrashantTripathi** [reply]
Maybe later in future when we'll get funded 😄

**Reply by mnsdojo** [reply]
because I'm working on a bhagwat reader  app

[https://github.com/mnsdojo/gita-reader](url)

**Comment by vsmm-world** [comment]
@PtPrashantTripathi Great work buddy !! appreciate your efforts
